### DISCUSSION TITLE
How to clear ui.upload filename?

### DISCUSSION THREAD

**Discussion by gbrandt** [zauberzeug/nicegui#185] [Q&A]
After a file is uploaded, I would like to reset the ui.upload component or at least clear the file name in the component.

Is there a way to do this?

**Comment by falkoschindler** [comment]
I managed to find a solution:

```python
from nicegui import ui

update = ui.upload()

async def reset():
    await update.page.run_javascript(f'document.getElementById({update.view.id}).reset()')

ui.button('reset', on_click=reset)

ui.run()
```

For convenience, we will add a `ui.upload.reset()` method with the next release.

**Reply by UStar987** [reply]
Hello, is there a way to let the program know that I executed an event to clear the component?

**Reply by BramleyAppleton** [reply]
you can now use the reset method:

```py
uploadfiles = ui.upload()
uploadfiles.reset()
```
